addons/l10n_es_edi_facturae/models/account_edi_format.py

```diff
@@ -0,0 +1,499 @@
+
+        items = []
+        totals = {
+            'total_gross_amount': 0.,
+            'total_general_discounts': 0.,
+            'total_general_surcharges': 0.,
+            'total_taxes_withheld': 0.,
+            'total_payments_on_account': 0.,
+            'amounts_withheld': 0.,
+        }
+        taxes = []
+        for line in invoice.invoice_line_ids:
+            invoice_line_values = EDI_XML_DICT_TEMPLATES['invoice_line'].copy()
+
+            total_price = line.quantity * line.price_unit * conv_rate  # This isn't a mistake
+            gross_amount = line.price_subtotal / conv_rate
+            discount = max(0., line.discount * total_price)
+            surcharge = max(0., - line.discount * total_price)  # + line.price_total - line.price_subtotal
+            totals['total_gross_amount'] += gross_amount
+            totals['total_general_discounts'] += discount
+            totals['total_general_surcharges'] += surcharge
+
+            taxes_output = [{
+                "TaxTypeCode": tax.l10n_es_edi_facturae_tax_type,
+                "TaxRate": f'{tax.amount:.3f}',
+                "TaxableBase": {
+                    'TotalAmount': fmt_in_cur(gross_amount),
+                    'EquivalentInEuros': fmt_in_eur(line.price_subtotal) if needs_equivalency else False,
+                },
+                "TaxAmount": {
+                    "TotalAmount": fmt_in_cur(compute_tax_amount(tax, line) * conv_rate),
+                    "EquivalentInEuros": fmt_in_eur(compute_tax_amount(tax, line)),
+                },
+            } for tax in line.tax_ids]
+
+            invoice_line_values.update({
+                'ItemDescription': line.name,
+                'Quantity': line.quantity,
+                'UnitOfMeasure': line.product_uom_id.l10n_es_edi_facturae_uom_code,
+                'UnitPriceWithoutTax': fmt_in_cur(line.price_unit),
+                'TotalCost': fmt_in_cur(total_price),
+                'DiscountsAndRebates': [
+                    {'DiscountReason': '', 'DiscountRate': f'{line.discount:.2f}', 'DiscountAmount': fmt_in_cur(discount)},
+                ] if discount != 0. else [],
+                'Charges': [
+                    {'ChargeReason': '', 'ChargeRate': f'{max(0, -line.discount):.2f}', 'ChargeAmount': fmt_in_cur(surcharge)},
+                ] if surcharge != 0. else [],
+                'GrossAmount': fmt_in_cur(gross_amount),
+                'TaxesOutputs': taxes_output,
+            })
+            items.append(invoice_line_values)
+            taxes += taxes_output
+        return items, taxes, totals
+
+    @api.model
+    def l10n_es_edi_facturae_export_facturae(self, invoice):
+        """
+        Produce the Facturae XML data for the invoice.
+
+        :param account.move invoice: The invoice to export.
+        :return tuple: (data needed to render the full template, data needed to render the signature template).
+        """
+
+        self_company = invoice.company_id
```

Comment:
```suggestion
        company = invoice.company_id
```
it's confusing... it looks like you are talking about `self.env.company`